lib/config/index.js

```diff
@@ -918,7 +918,8 @@ function getHostnameSafe() {
     return _hostname
   }
   try {
-    _hostname = os.hostname()
+    const dynoName = process.env.DYNO
```

Comment:
The host name needs to pull from process.env.DYNO only if the configuration value is true.  You can see that here https://github.com/newrelic/newrelic-ruby-agent/blob/dev/lib/new_relic/agent/hostname.rb#L12.  To add a configuration item in this repo you can define it in `lib/config/default.js`.  

It'll need to be like this
```js
heroku: {
  use_dyno_names: {
    default: true,
    formatter: boolean
  }
}
```

Also this PR lacks test updates. Please make updates to existing tests to assert this new configuration item and hostname update when config is true and process.env.DYNO exists

Reply:
I've added tests and the default config.